src/Shared/Http2cat/Http2Utilities.cs

```diff
@@ -4,34 +4,49 @@
 using System;
 using System.Buffers;
 using System.Buffers.Binary;
-using System.Collections.Concurrent;
 using System.Collections.Generic;
 using System.IO;
 using System.IO.Pipelines;
 using System.Linq;
 using System.Net.Http;
 using System.Net.Http.HPack;
 using System.Text;
+using System.Threading;
 using System.Threading.Tasks;
 using Microsoft.AspNetCore.Connections;
 using Microsoft.AspNetCore.Http;
-using Microsoft.AspNetCore.Server.Kestrel.Core.Internal.Http;
 using Microsoft.AspNetCore.Server.Kestrel.Core.Internal.Http2;
 using Microsoft.AspNetCore.Server.Kestrel.Core.Internal.Infrastructure;
+using Microsoft.Extensions.Logging;
 using Microsoft.Net.Http.Headers;
 
-namespace http2cat
+namespace Microsoft.AspNetCore.Http2Cat
```

Comment:
Sucks that we have so much duplicated code between this and Http2TestBase: https://github.com/aspnet/AspNetCore/blob/master/src/Servers/Kestrel/test/InMemory.FunctionalTests/Http2/Http2TestBase.cs

Reply:
It might be possible to re-unify them, but we'd have to work out how to toggle between sockets and in-memory. This PR is already big enough, we'll follow up on that later.